llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -1210,6 +1211,41 @@ static SDValue performSHLCombine(SDNode *N, SelectionDAG &DAG,
                      DAG.getConstant(SMSize, DL, MVT::i32));
 }
 
+static SDValue performSignExtendCombine(SDNode *N, SelectionDAG &DAG, TargetLowering::DAGCombinerInfo &DCI,const MipsSubtarget &Subtarget) {
+  if (DCI.Level != AfterLegalizeDAG) {
+    return SDValue();
+  }
+
+  if (!Subtarget.isGP64bit()) {
+    return SDValue();
+  }
+
+  SDValue N0 = N->getOperand(0);
+  EVT VT = N->getValueType(0);
+
+  if (N0->getNumOperands() != 2) {
+    return SDValue();
+  }
+
+  // Pattern match XOR.
+  //  $dst = (sign_extend (xor (trunc $src, i32), -1), i64)
+  //  => $dst = (xor ($src, -1), i64)
+  if (N0.getOpcode() == ISD::XOR &&
+      N0.getOperand(0).getOpcode() == ISD::TRUNCATE &&
+      N0.getOperand(1).getOpcode() == ISD::Constant) {
+    SDValue TruncateOperand = N0.getOperand(0).getOperand(0);
+    if (VT == MVT::i64 && VT == TruncateOperand->getValueType(0)) {
```

Comment:
This should be `TruncateOperand.getValueType()`. You don't that result 0 of the `TruncateOperand` node is being used. The result number is stored in the SDValue so you should use it.